src/lib/sw.js

```diff
@@ -192,13 +192,18 @@ workboxRouting.registerRoute(normalMatch, async ({url}) => {
   }
   const partial = await response.json();
 
-  // Our target browsers all don't mind if we just place <title> in the middle of the document.
-  // This is far simpler than trying to find the right place in <head>.
   const meta = partial.offline ? `<meta name="offline" value="true" />` : '';
-  const output = layoutTemplate.replace(
-    '%_CONTENT_REPLACE_%',
-    meta + `<title>${partial.title || ''}</title>` + partial.raw,
-  );
+  const title = `<title>${partial.title || 'web.dev'}</title>`;
+  const rssHref = partial.rss || '/feed.xml';
+  const rssTitle =
+    partial.rss !== '/feed.xml'
+      ? `${partial.title} on web.dev`
+      : 'web.dev feed';
+  const rss = `<link rel="alternate" href="${rssHref}" type="application/atom+xml" data-title="${rssTitle}" />`;
+
+  const output = layoutTemplate
+    .replace('<title>web.dev</title>', `${meta}\n${title}\n${rss}`)
```

Comment:
Would it be safer to use a regex to replace the first title tag, regardless of its text content? My thinking is that if we ever changed this text to `<title>web.dev | learn new things!</title>` then this would (silently?) start failing.

@samthor should double check this my regex here, but I was thinking something like:

`\<title\>.*\<\/title\>`

Reply:
This is also a really good reason to write some service worker tests, to catch those potential failures :D